core/server/data/migrations/versions/3.0.0/1-remove-user-ghost-auth-columns.js

```diff
@@ -0,0 +1,78 @@
+const common = require('../../../../lib/common');
+const commands = require('../../../schema').commands;
+
+const createLog = type => msg => common.logging[type](msg);
+
+function createColumnMigrations(migrations) {
+    return function columnMigrations({transacting}) {
+        return Promise.each(migrations, function ({table, column, dbIsInCorrectState, operation, operationVerb, columnDefinition}) {
+            return transacting.schema.hasColumn(table, column)
+                .then(dbIsInCorrectState)
+                .then((isInCorrectState) => {
+                    const log = createLog(isInCorrectState ? 'warn' : 'info');
+
+                    log(`${operationVerb} ${table}.${column}`);
+
+                    if (!isInCorrectState) {
+                        return operation(table, column, transacting, columnDefinition);
+                    }
+                });
+        });
+    };
+}
+
+module.exports.up = createColumnMigrations([
+    {
+        table: 'users',
+        column: 'ghost_auth_access_token',
+        dbIsInCorrectState(columnExists) {
+            return columnExists === false;
+        },
+        operation: commands.dropColumn,
+        operationVerb: 'Removing'
+    },
+    {
+        table: 'users',
+        column: 'ghost_auth_id',
+        dbIsInCorrectState(columnExists) {
+            return columnExists === false;
+        },
+        operation: commands.dropColumn,
+        operationVerb: 'Removing'
+    }
+]);
+
+module.exports.down = createColumnMigrations([
+    {
+        table: 'users',
+        column: 'ghost_auth_access_token',
+        dbIsInCorrectState(columnExists) {
+            return columnExists === true;
+        },
+        operation: commands.addColumn,
+        operationVerb: 'Adding',
+        columnDefinition: {
+            type: 'string',
+            nullable: true,
+            maxLength: 32
```

Comment:
The down migration doesn't seem to have taken `maxLength` bit into account, state after migration back to 2.x.x from 3.0.0 is:
```
| ghost_auth_access_token | varchar(191)  | YES  |     | NULL    |       |
| ghost_auth_id           | varchar(191)  | YES  |     | NULL    |       |
```
originally was:
```
| ghost_auth_access_token | varchar(32)   | YES  |     | NULL    |       |
| ghost_auth_id           | varchar(24)   | YES  |     | NULL    |       |
```

Another bit I'm thinking is the positioning of the columns. They are being inserted back at the end of the table instead of into their original position after `slug` column :thinking: 

It's a down migration, so is not too critical, but probably needs to be solved if we want to reuse this code in these things matter in future :)

Reply:
@gargol Fixed the issue with `maxlength` being set wrong, good catch. The issue was camelCase vs lowercase in setting the property. 

For the positioning of columns, knex does has one [method](http://knexjs.org/#Schema-after) to ensure its inserted after a certain column, though its limited to mysql. This is actually gonna be an issue for all adds we do in `down` migration even in future, so probably something to ignore for now as it shouldn't make much difference in this case?

Reply:
> This is actually gonna be an issue for all adds we do in down migration even in future, so probably something to ignore for now as it shouldn't make much difference in this case?

Yes, agree it's not something to spend time on atm :+1: